libs/payments/cart/src/lib/checkout.service.ts

```diff
@@ -701,4 +703,49 @@ export class CheckoutService {
 
     return upgradedSubscription;
   }
+
+  async determineCheckoutAmount({
+    eligibility,
+    customer,
+    priceId,
+    currency,
+    taxAddress,
+  }: {
+    eligibility: SubscriptionEligibilityResult;
+    priceId: string;
+    currency: string;
+    taxAddress: TaxAddress;
+    customer?: StripeCustomer;
+  }) {
+    eligibility.subscriptionEligibilityResult
+    if (eligibility.subscriptionEligibilityResult === EligibilityStatus.UPGRADE) {
+      assert(
+        'fromPrice' in eligibility,
+        'fromPrice not present for upgrade cart'
+      );
+      assert(customer, 'Customer is required for upgrade')
```

Comment:
Instead of the second argument being a string, if it is an `Error` subclass, then that error will be thrown instead of `AssertionError`. This feels like it would build nicely on the error refactoring work we did before in increasing the granularity of our logs https://nodejs.org/api/assert.html#assertokvalue-message